gobblin-azkaban/src/main/java/gobblin/azkaban/AzkabanJobLauncher.java

```diff
@@ -40,50 +42,59 @@
   private static final String HADOOP_TOKEN_FILE_LOCATION = "HADOOP_TOKEN_FILE_LOCATION";
   private static final String MAPREDUCE_JOB_CREDENTIALS_BINARY = "mapreduce.job.credentials.binary";
 
-  private final Properties properties;
+  private final Closer closer = Closer.create();
   private final JobLauncher jobLauncher;
+  private final JobListener jobListener = new EmailNotificationJobListener();
 
   public AzkabanJobLauncher(String jobId, Properties props)
       throws Exception {
     super(jobId, LOG);
 
-    this.properties = new Properties();
-    this.properties.putAll(props);
+    Properties properties = new Properties();
```

Comment:
Just some food for thought, but should we continue to rely on the Java `Properties` class. It has some weird behavior. Check out the Java Docs: http://docs.oracle.com/javase/7/docs/api/java/util/Properties.html

Specifically the part about the `put` and `putAll` method. The methods directly modify the `HashTable` that backs the `Properties` object, and the methods allow a user to add non-`String` value types.

I've also never been a fan of the `new Properties(Properties defaults)` constructor. It's confusing because it looks like a copy constructor, but in reality it sets property "defaults", which don't get returned by functions such as `keySet()`.